### PR TITLE
Added assignment code to flesh out skeleton

### PR DESCRIPTION
**Pull request by MangoPicante**

### CODE REVIEW

File: src/main/java/StringManipulation.java [hunk 1]
```diff
@@ -1,32 +1,99 @@
...
     @Override
     public int count() {
-        return 0;
+        StringTokenizer st = new StringTokenizer(string, " ");
+        return st.countTokens();
     }
 
     @Override
     public String removeNthCharacter(int n, boolean maintainSpacing) {
-        return null;
```
Comment (birdhead9) [inline src/main/java/StringManipulation.java:19]: I think you forgot to delete this as you pasted your code into this method . :P

File: src/main/java/StringManipulation.java [hunk 2]
```diff
@@ -1,32 +1,99 @@
...
     }
 
     @Override
     public void setString(String string) {
+        this.string = string;
     }
 
     @Override
     public int count() {
-        return 0;
```
Comment (birdhead9) [inline src/main/java/StringManipulation.java:14]: Also here, you forgot to delete as your pasted your code into the skeleton code.

File: src/main/java/StringManipulation.java [hunk 3]
```diff
@@ -1,32 +1,99 @@
...
+        int i = 0;
+        while (st.hasMoreTokens()) {
+            substrings[i++] = st.nextToken();
+        }
+        // return copy of array with only requested words
+        return Arrays.copyOfRange(substrings,startWord - 1,endWord);
     }
 
     @Override
-    public String restoreString(int[] indices) {
```
Comment (birdhead9) [inline src/main/java/StringManipulation.java:28]: You have two methods named restoreString .XD

File: src/main/java/StringManipulation.java [hunk 4]
```diff
@@ -1,32 +1,99 @@
...
+// Sean Michael CS410
+
+import java.util.Arrays;
+import java.util.StringTokenizer;
 
+public class StringManipulation implements StringManipulationInterface {
+    String string;
     @Override
     public String getString() {
         return null;
```
Comment (birdhead9) [inline src/main/java/StringManipulation.java:10]: I noticed you have just return null here.

File: src/main/java/StringManipulation.java [hunk 5]
```diff
@@ -1,32 +1,99 @@
-public class StringManipulation implements StringManipulationInterface {
```
Comment (birdhead9) [inline src/main/java/StringManipulation.java:1]: You have two "public class String....implements StringManipulationInter...{ ."

File: src/main/java/StringManipulation.java [hunk 6]
```diff
@@ -1,32 +1,99 @@
...
+                else sb.deleteCharAt(i);
+            }
+            return sb.toString();
+        }
+
     }
 
     @Override
-    public String[] getSubStrings(int startWord, int endWord) {
-        return null;
```
Comment (birdhead9) [inline src/main/java/StringManipulation.java:24]: There is two getSubstrings methods here

File: src/main/java/StringManipulation.java (hunk 5 above)
Comment (MangoPicante) [reply src/main/java/StringManipulation.java:1]: The - before the line indicates that it was removed from the last commit

File: src/main/java/StringManipulation.java (hunk 2 above)
Comment (MangoPicante) [reply src/main/java/StringManipulation.java:14]: The - before the line indicates that it was removed from the last commit

File: src/main/java/StringManipulation.java (hunk 1 above)
Comment (MangoPicante) [reply src/main/java/StringManipulation.java:19]: The - before the line indicates that it was removed from the last commit

File: src/main/java/StringManipulation.java (hunk 6 above)
Comment (MangoPicante) [reply src/main/java/StringManipulation.java:24]: The - before the line indicates that it was removed from the last commit

File: src/main/java/StringManipulation.java (hunk 3 above)
Comment (MangoPicante) [reply src/main/java/StringManipulation.java:28]: The - before the line indicates that it was removed from the last commit

File: src/test/java/StringManipulationTest.java [hunk 7]
```diff
@@ -19,130 +21,179 @@ public void tearDown() {
...
     }
 
-    @Test
+    @Test // Tests standard usage of restore string
     public void testRestoreString1()
     {
         manipulatedstring.setString("art");
-        int [] array;
-        array=new int[]{1,0,2};
-        String restoreString = manipulatedstring.restoreString(array);
```
Comment (birdhead9) [inline src/test/java/StringManipulationTest.java:117]: I think there's a lot more copy and paste error, as there are two sets of lines that are similar.

File: src/test/java/StringManipulationTest.java [hunk 8]
```diff
@@ -19,130 +21,179 @@ public void tearDown() {
...
+        manipulatedstring.setString("art");
+        assertThrows(IndexOutOfBoundsException.class, () -> manipulatedstring.restoreString(new int[] {4, 5, 6}));
 
     }
 
-    @Test
+    @Test // Tests illegal argument exception for too many indices passed
     public void testRestoreString3()
     {
-        fail("Not yet implemented");
```
Comment (birdhead9) [inline src/test/java/StringManipulationTest.java:131]: This test will only run the fail part.

File: src/test/java/StringManipulationTest.java [hunk 9]
```diff
@@ -19,130 +21,179 @@ public void tearDown() {
...
+        manipulatedstring.setString("art");
+        assertThrows(IllegalArgumentException.class, () -> manipulatedstring.restoreString(new int[] {1, 2, 3, 4}));
 
     }
 
-    @Test
+    @Test // Tests empty string restoration
     public void testRestoreString4()
     {
-        fail("Not yet implemented");
```
Comment (birdhead9) [inline src/test/java/StringManipulationTest.java:138]: fail error here. need to delete.

File: src/test/java/StringManipulationTest.java [hunk 10]
```diff
@@ -19,130 +21,179 @@ public void tearDown() {
...
 
     }
 
-    @Test
+    @Test // Tests single char string restoration
     public void testRestoreString5()
     {
-        fail("Not yet implemented");
+        manipulatedstring.setString(" ");
+        String restoreString = manipulatedstring.restoreString(new int[] {1});
```
Comment (birdhead9) [inline src/test/java/StringManipulationTest.java:195]: There is an index out of bounds here.

File: src/test/java/StringManipulationTest.java (hunk 7 above)
Comment (MangoPicante) [reply src/test/java/StringManipulationTest.java:117]: The - before the line indicates that it was removed from the last commit

File: src/test/java/StringManipulationTest.java (hunk 8 above)
Comment (MangoPicante) [reply src/test/java/StringManipulationTest.java:131]: The - before the line indicates that it was removed from the last commit

File: src/test/java/StringManipulationTest.java (hunk 9 above)
Comment (MangoPicante) [reply src/test/java/StringManipulationTest.java:138]: The - before the line indicates that it was removed from the last commit

File: src/test/java/StringManipulationTest.java [hunk 11]
```diff
@@ -19,130 +21,179 @@ public void tearDown() {
...
+    @Test // Tests replacing every character with a space
     public void testRemoveNthCharacter7() {
-        fail("Not yet implemented");
+        manipulatedstring.setString("123456789");
+        assertEquals("         ", manipulatedstring.removeNthCharacter(1,true),manipulatedstring.getString());
     }
 
-    @Test
-    public void testGeSubStrings1() {
-        manipulatedstring.setString("This is my string");
```
Comment (birdhead9) [inline src/test/java/StringManipulationTest.java:83]: random method in between

Comment (MangoPicante) [reply src/test/java/StringManipulationTest.java:83]: The - before the line indicates that it was removed from the last commit

**Review by birdhead9 — CHANGES_REQUESTED** [review]
Hello, Your code is really nice and straightforward. I just noticed that there were a lot of copy and paste errors. Might want to address those to make sure your code works without those. Also, in your test cases, you didn't remove the template "fail(not yet implemented)" in a lot of the test cases, so your test cases will not run, as it will just stop after that fail line.

**Review by birdhead9 — APPROVED** [review]

**Pull request closed without merging**